src/main/java/org/aion/wallet/account/AccountManager.java

```diff
@@ -78,39 +81,59 @@ private void registerEventBusConsumer() {
         EventBusFactory.getBus(SettingsEvent.ID).register(this);
     }
 
-    public String createAccount(final String password, final String name) {
+    public String createMasterAccount(final String password, final String name) throws ValidationException {
         final StringBuilder mnemonicBuilder = new StringBuilder();
         final byte[] entropy = new byte[Words.TWELVE.byteLength()];
         new SecureRandom().nextBytes(entropy);
         new MnemonicGenerator(English.INSTANCE).createMnemonic(entropy, mnemonicBuilder::append);
         final String mnemonic = mnemonicBuilder.toString();
-        final byte[] seed = getNewAccountSeed(mnemonic);
-        final ECKey ecKey = new SeededECKeyEd25519(seed);
 
-        final String address = Keystore.create(password, ecKey);
-        if (address.equals("0x")) {
-            log.error("An exception occurred while creating the new account");
+        final AccountDTO account = processMasterAccount(mnemonic, password);
+        if (account == null) {
             return null;
-        } else {
-            final byte[] fileContent = keystoreFormat.toKeystore(ecKey, password);
-            final AccountDTO account = createAccountWithPrivateKey(address, ecKey.getPrivKeyBytes());
-            if (account == null) {
-                return null;
-            } else {
-                account.setName(name);
-                processAccountAdded(account, fileContent, true);
-                storeAccountName(address, name);
-                if (isAccountAlreadyImported(account)) {
-                    return null;
-                }
-                return mnemonic;
-            }
+        }
+        account.setName(name);
+        storeAccountName(account.getPublicAddress(), name);
+        return mnemonic;
+    }
+
+    public void importMasterAccount(final String mnemonic, final String password) throws ValidationException {
+        try {
+            processMasterAccount(mnemonic, password);
+        } catch (final Exception e) {
+            throw new ValidationException(e);
+        }
+    }
+
+    private AccountDTO processMasterAccount(String mnemonic, String password) throws ValidationException {
```

Comment:
`final`
in IDEA you can set auto-final for param/variable generation from Settings->CodeStyle->Java->CodeGenerarion

Reply:
Method is `private` so a `final` modifier would be pointless here.